ykman/cli/util.py

```diff
@@ -67,14 +67,11 @@ def inner(ctx, param, val):
     return wrap
 
 
-def click_skip_on_help(f):
+def click_postpone_exection(f):
     @functools.wraps(f)
     def inner(*args, **kwargs):
         ctx = click.get_current_context()
-        for arg in ctx.help_option_names:
-            if arg in sys.argv:
-                return
-        f(*args, **kwargs)
+        ctx.obj['dev'].on_resolves.append(lambda: f(*args, **kwargs))
```

Comment:
This should probably be encapsulated with an `on_resolve` method or the like instead. Then that could also call the argument immediately if it's already resolved.

(I kind of want to call it `.then(lambda...)` like Promises in JS, but that's probably not a very good idea...)